client/blocks/signup-form/index.jsx

```diff
@@ -194,19 +207,37 @@ class SignupForm extends Component {
 	}
 
 	sanitize = ( fields, onComplete ) => {
-		const sanitizedEmail = this.sanitizeEmail( fields.email ),
-			sanitizedUsername = this.sanitizeUsername( fields.username );
-
-		if ( fields.email !== sanitizedEmail || fields.username !== sanitizedUsername ) {
+		const sanitizedEmail = this.sanitizeEmail( fields.email );
+		const sanitizedFirstName = this.sanitizeName( fields.firstName );
+		const sanitizedLastName = this.sanitizeName( fields.lastName );
+		const sanitizedUsername = this.sanitizeUsername( fields.username );
+
+		if (
+			fields.email !== sanitizedEmail ||
+			fields.firstName !== sanitizedFirstName ||
+			fields.lastName !== sanitizedLastName ||
+			fields.username !== sanitizedUsername
+		) {
 			onComplete( {
 				email: sanitizedEmail,
+				firstName: sanitizedFirstName,
+				lastName: sanitizedLastName,
 				username: sanitizedUsername,
+				password: fields.password,
 			} );
 		}
 	};
 
 	validate = ( fields, onComplete ) => {
-		wpcom.undocumented().validateNewUser( fields, ( error, response ) => {
+		const data = pick( fields, [ 'email', 'username', 'password' ] );
+
+		if ( this.props.useFullName ) {
```

Comment:
Is there a case where you may want a full name AND username?

I'd suggest using 2 explicit props for turning these on and off. :)

Reply:
I broke up `useFullName` into `displayNameInput` and `displayUsernameInput`.